### PR TITLE
Add test matrix and bump OTP version

### LINKED ISSUE #146: Build matrix release
based on this comment: https://github.com/lexical-lsp/lexical/pull/111#discussion_r1189197244

> We need to build a matrix like this:

```
Erlang: Elixir
24 1.13, 1.14
25 1.13, 1.14
26 1.13, 1.14, (1.15)
```

### PR DESCRIPTION
**Pull request by wingyplus**
Closes #146

### CODE REVIEW

File: .github/workflows/elixir.yml [hunk 1]
```diff
@@ -18,15 +18,52 @@ permissions:
...
+
+      - name: Install dependencies
+        run: |
+          mix deps.get
+
+      # Step: Check that the checked in code has already been formatted.
+      # This step fails if something was found unformatted.
+      # Customize this step as desired.
+      - name: Check Formatting
+        run: mix format --check-formatted
```
Comment (scohen) [inline .github/workflows/elixir.yml:56]: Why this change? We have a precommit hook and testing task that should prevent unformatted code from being merged into main. 

Also, I don't think you need to fetch and compile deps just to check formatting

Comment (wingyplus) [reply .github/workflows/elixir.yml:56]: @scohen I found that format style diff on Elixir version 1.13 T_T (https://github.com/lexical-lsp/lexical/actions/runs/4938201783/jobs/8827646922). So to avoid this I extract to a new job.

Comment (wingyplus) [reply .github/workflows/elixir.yml:56]: Or add some boolean in matrix (like `check-format: true | false`) and `if` at this step would be better?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Set up Elixir
         uses: erlef/setup-beam@v1
         with:
-          otp-version: "25.2.1"
+          otp-version: "25.3.2"
```
[the remark this answers is not in the document]
Comment (wingyplus) [reply .github/workflows/release.yml:17]: I think it should be. Prefer edit in this PR or open a new one?

**Comment by wingyplus** [comment]
I'll work again tomorrow. The action is unstable now (https://www.githubstatus.com/)

File: .github/workflows/release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (scohen) [reply .github/workflows/release.yml:17]: Do it here, though we should set it to the lowest value we support. 
1.13 / erl 24?

[the remark this answers is not in the document]
Comment (wingyplus) [reply .github/workflows/release.yml:17]: I think we should verify formatting on elixir 1.14 because we use 1.14 for development (reference from .tools-version).

File: .github/workflows/elixir.yml (hunk 1 above)
Comment (scohen) [reply .github/workflows/elixir.yml]: how about we just format it so that 1.13 doesn't complain?

File: .github/workflows/release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (scohen) [reply .github/workflows/release.yml:17]: If we're going to support 1.13, we need to develop on that. Using a later version will only cause us pain.

**Comment by wingyplus** [comment]
Just note on Elixir 1.13 issues:

1. Mix format diff on config/prod.exs (Fixed in 
2. Dialyzer failure on this version 
```
lib/mix/tasks/lsp/data_model.ex:82:no_return
Function collect_references/3 has no local return.
________________________________________________________________________________
lib/mix/tasks/lsp/data_model.ex:82:opaque_match
Attempted to pattern match against the internal structure of an opaque term.

Type:
MapSet.t(_)

Pattern:
%MapSet{}

This breaks the opaqueness of the term.

________________________________________________________________________________
lib/mix/tasks/lsp/data_model.ex:86:opaque_match
Attempted to pattern match against the internal structure of an opaque term.

Type:
MapSet.t(_)

Pattern:
%MapSet{}

This breaks the opaqueness of the term.

________________________________________________________________________________
lib/mix/tasks/lsp/mappings/generate.ex:78:call_to_missing
Call to missing or private function Keyword.replace_lazy/3.
________________________________________________________________________________
done (warnings were emitted)
Halting VM with exit status 2
``` 
3. Found warning on compiling proto apps due to  `Keyword.replace_lazy` introduced in Elixir 1.14

File: config/prod.exs [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (wingyplus) [inline config/prod.exs:1]: Elixir 1.13 format like this. On Elixir 1.14, it removed this line.

**Comment by wingyplus** [comment]
@scohen I found a lot of test failures on Elixir 1.13 https://github.com/lexical-lsp/lexical/actions/runs/4950410802/jobs/8853977395?pr=152.  Need to fix the test first before adding Elixir 1.13 to the CI.
